testrunner.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+#*************************GO-LICENSE-START********************************
+# Copyright 2015 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#*************************GO-LICENSE-END**********************************
+
+function exit_on_error
+{
+   echo "$1" 1>&2
+   exit 1
+}
+
+sudo apt-get -y update
+sudo apt-get -y install openjdk-7-jre unzip git ruby rubygems || exit_on_error "failed to install necessary dependencies"
+sudo gem install json rake || exit_on_error "failed to install json and rake"
+
+sudo apt-get -y install ruby-nokogiri
```

Comment:
Please don't mix installing gems via `gem install` and `apt-get`. Use one or the other, not both.

It's even better if you use use `bundler` instead, and this works great if you have a `Gemfile` checked in locally, that way the basic packages you'd need to install via `apt-get` would be the JRE, ruby, rubygems and bundler.

Everything else ruby comes in via `bundler`.